src/pages/index.js

```diff
@@ -2,76 +2,215 @@ import "./index.css";
+  });
+
+  itemSection.renderItems(items);
 });
-itemSection.renderItems();
 
-const addCardModal = new PopupWithForm(
-  "#profile-add-card",
-  handleCardAddSubmit
+const userInfo = new UserInfo(
+  ".profile__title",
+  ".profile__description",
+  ".profile__image"
 );
-addCardModal.setEventListeners();
 
-function handleImageClick(name, link) {
-  previewImageModal.open(name, link);
+function renderCard(card) {
+  const cardElement = createCard(card);
+  itemSection.addItem(cardElement);
 }
 
-function createCard(cardData) {
-  return new Card(cardData, cardTemplateSelector, handleImageClick).getView();
+api
+  .getUserInfo()
+  .then((userData) => {
+    userInfo.setUserInfo(userData);
+    // a = true
+  })
+  .catch((err) => {
+    console.error(err);
+  });
+
+const deleteConfirm = new PopupWithConfirmation("#card-delete-modal");
+const handleDeleteConfirm = (card) =>
+  api
+    .deleteCard(card._id)
+    .then(() => {
+      console.log(card);
+      card.remove();
+    })
+    .catch((err) => {
+      console.log(err);
+    });
+deleteConfirm.setEventListeners();
+
+function handleDeleteClick(card) {
+  // Set the submit action for the confirmation popup
+  console.log(card._id);
+  // why don't we open confirm modal first?
+  deleteConfirm.setSubmitAction(() => {
+    console.log(deleteConfirm);
+    handleDeleteConfirm(card);
+    deleteConfirm.close();
+  });
 }
 
-function renderCard(cardData) {
-  const cardElement = createCard(cardData);
-  itemSection.addItem(cardElement);
+function handleLikeClick(card) {
+  api
+    .toggleCardLike(card._id, card.isLiked)
+    .then((_isLiked) => {
+      // console.log("RESPONSE");
+      // console.log(_isLiked);
+    })
+    .catch((err) => {
+      console.error(err);
+    });
+}
+
+function handleChangePicture(avatar) {
+  changePictureModal.setEventListeners();
+  console.log(avatar);
+  this._buttonElement = this._popupForm.querySelector(".modal__button");
+  this._buttonElement;
+  changePictureModal.renderLoading(true);
+  api
+    .setUserAvatar(avatar)
+    .then((data) => {
+      changePictureModal.renderLoading(false);
+      console.log(data.avatar);
+      userInfo.setUserAvatar(data.avatar);
+    })
+    .catch((err) => {
+      console.error(err);
+    });
+
+  changePictureModal.close();
 }
 
-function handleProfileEditSubmit(inputValues) {
-  userInfo.setUserInfo(inputValues.title, inputValues.description);
+const changePictureModal = new PopupWithForm(
+  "#profile-change-picture",
+  handleChangePicture
+);
+changePictureModal.setEventListeners();
+const pictureFormValidator = new FormValidator(config, pictureChangeForm);
+pictureFormValidator.enableValidation();
+profilePicture.addEventListener("click", () => {
+  console.log("not working");
+  pictureFormValidator. resetValidation();
+  changePictureModal.open();
+});
+
+const editProfileModal = new PopupWithForm(
+  "#profile-edit-modal",
+  (userData) => {
+    editProfileModal.renderLoading(true);
```

Comment:
**Could be improved**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```